eli5/keras/explain_prediction.py

```diff
@@ -0,0 +1,226 @@
+from .gradcam import gradcam, gradcam_backend
+
+
+DESCRIPTION_KERAS = """Grad-CAM visualization for image classification; 
+output is explanation object that contains input image 
+and heatmap image for a target.
+"""
+
+# note that keras.models.Sequential subclasses keras.models.Model
+@explain_prediction.register(Model)
+def explain_prediction_keras(estimator, # type: Model
+                             doc, # type: np.ndarray
+                             target_names=None,
+                             targets=None, # type: Optional[list]
+                             layer=None, # type: Optional[Union[int, str, Layer]]
+                            ):
+    # type: (...) -> Explanation
+    """
+    Explain the prediction of a Keras image classifier.
+
+    We make two explicit assumptions
+        * The input is images.
+        * The model's task is classification, i.e. final output is class scores.
+
+    See :func:`eli5.explain_prediction` for more information about the ``estimator``,
+    ``doc``, ``target_names``, and ``targets`` parameters.
+
+    
+    :param keras.models.Model estimator:
+        Instance of a Keras neural network model, 
+        whose predictions are to be explained.
+
+
+    :param numpy.ndarray doc:
+        An input image as a tensor to ``estimator``, 
+        from which prediction will be done and explained.
+
+        For example a ``numpy.ndarray``.
+
+        The tensor must be of suitable shape for the ``estimator``. 
+
+        For example, some models require input images to be 
+        rank 4 in format `(batch_size, dims, ..., channels)` (channels last)
+        or `(batch_size, channels, dims, ...)` (channels first), 
+        where `dims` is usually in order `height, width`
+        and `batch_size` is 1 for a single image.
+
+        Check ``estimator.input_shape`` to confirm the required dimensions of the input tensor.
+
+
+        :raises TypeError: if ``doc`` is not a numpy array.
+        :raises ValueError: if ``doc`` shape does not match.
+
+    :param target_names:         
+        *Not Implemented*.
+        Names for classes in the final output layer.
+    :type target_names: list, optional
+
+    :param targets:
+        Prediction ID's to focus on.
+
+        *Currently only the first prediction from the list is explained*. 
+        The list must be length one.
+
+        If None, the model is fed the input image and its top prediction 
+        is taken as the target automatically.
+
+
+        :raises ValueError: if ``targets`` is a list with more than one item.
+        :raises TypeError: if ``targets`` is not list or None.
+    :type targets: list[int], optional
+
+    :param layer:
+        The activation layer in the model to perform Grad-CAM on,
+        a valid keras layer name, layer index, or an instance of a Keras layer.
+        
+        If None, a suitable layer is attempted to be retrieved. 
+        See :func:`eli5.keras._search_layer_backwards` for details.
```

Comment:
I think if we're documenting a function, it makes sense to make it public - or just document the  behavior without mentioning  a function.

Reply:
Good catch. Forgot that I made the function private.

https://github.com/TeamHG-Memex/eli5/pull/315/commits/bcaf7ca5724189b88f0529a0ca85aa1ccf860bd9